### PR TITLE
Allow latest `psr/*` dependencies, drop PHP 7.3 support

### PR DESCRIPTION
**Pull request by geerteltink**
This PR allow latest psr dependencies.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,27 +20,30 @@
...
             "config-provider": "Mezzio\\Authorization\\Acl\\ConfigProvider"
         }
     },
     "require": {
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
-        "laminas/laminas-permissions-acl": "^2.6",
-        "mezzio/mezzio-authorization": "^1.0",
-        "mezzio/mezzio-router": "^3.0",
-        "psr/container": "^1.0",
+        "php": "^7.4 || ~8.0.0 || ~8.1.0",
```
Comment (froschdesign) [inline composer.json:34]: You remove the support for version 7.3 here but the topic of this pull request is the update of the PSR support.This results in wrong releases notes!

I see two options: write a milestone description or create a separate pull request.

Example for wrong release notes: https://github.com/mezzio/mezzio-helpers/releases/tag/5.8.0

Thanks in advance!

Comment (geerteltink) [reply composer.json:34]: Single merge requests is hard. It's all related:

- PHP 7.3 fails for most packages. 
- Some need an update for coding standards to 2.3.
- Psalm has new checks after you update something, which need  to be addressed.

How do I write the milestone descriptions?

Comment (froschdesign) [reply composer.json:34]: Go to issues then milestones or click on any link to a milestone and edit the milestone.

https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/creating-and-editing-milestones-for-issues-and-pull-requests

Comment (Ocramius) [reply composer.json:34]: Adjusted title - good enough for release notes.

**Review by Ocramius — APPROVED** [review]
Thanks @geerteltink!

**Pull request merged**